[#74682] Update Jira migration docs (#23087)
opf/openproject: docs/installation-and-operations/jira-migration/custom-fields/README.md

---
sidebar_navigation:

>   title: Custom fields

Reviewer: @as-op maybe we could be more specific here? "Custom fields migration" instead of "Custom fields"?

Author: Done!

description: Migration of Jira custom fields to OpenProject during a Jira Data Center migration. Supported field types, edge cases, and limitations.
keywords: Jira custom fields, OpenProject custom fields, field type mapping, field contexts, unsupported field types
---

# Custom fields migration from Jira to OpenProject

> The Jira Migrator automatically detects custom fields that are actually used in your imported issues and 

Reviewer: ```suggestion
The Jira Migrator automatically detects custom fields that are actually used in your imported issues, and 
```

creates matching OpenProject custom fields for them. 

> Fields that exist in Jira but carry no values in any imported issue are ignored.

Reviewer: ```suggestion
Fields that exist in Jira but contain no values in any imported issue are ignored.
```

## Currently unsupported field types

Jira custom field types not listed above are skipped. This includes (but is not limited to):

- Datetime fields
- Version fields
- Sprint assignment
- Epic links
- Story points
- Third-party plugin fields

If a field is skipped, its values are not imported and no OpenProject custom field is created for it.

## Field type details and edge cases

### Checkboxes

Jira checkbox fields (`multicheckboxes`) behave differently depending on how many options the field has in the imported data:

- **Single option**: The field becomes an OpenProject **Boolean** (yes/no) custom field named `<FieldName> - <OptionValue>`. It is `true` if the issue had that option checked.
- **Multiple options**: The field becomes a multi-value **List** custom field containing all checked values.

If a checkbox field has different option sets across projects (via Field Contexts), both rules above apply independently per context group. 
A single Jira checkbox field can result in multiple OpenProject custom fields.

### Cascading select

> Jira cascading select fields have two import modes depending on your OpenProject plan:

Reviewer: ```suggestion
Jira cascading select fields have two import modes depending on your OpenProject edition:
```

> - **Enterprise plan (custom field hierarchies enabled)**: The field is imported as an OpenProject **Hierarchy** custom field. 

Reviewer: ```suggestion
- **Enterprise edition (custom field hierarchies are enabled)**: The field is imported as an OpenProject **Hierarchy** custom field. 
```

  The full parent-child tree is preserved. A selected value of `Animals > Cat` is stored as the `Cat` item under the `Animals` parent.

> - **Community/Basic plan**: The field is imported as a **List** custom field. 

Reviewer: ```suggestion
- **Community edition**: The field is imported as a **List** custom field. 
```
